Testing/nrrd/tloadchunk.c

```diff
@@ -0,0 +1,143 @@
+/*
+  Teem: Tools to process and visualize scientific data and images             .
+  Copyright (C) 2013, 2012, 2011, 2010, 2009  University of Chicago
+  Copyright (C) 2008, 2007, 2006, 2005  Gordon Kindlmann
+  Copyright (C) 2004, 2003, 2002, 2001, 2000, 1999, 1998  University of Utah
+
+  This library is free software; you can redistribute it and/or
+  modify it under the terms of the GNU Lesser General Public License
+  (LGPL) as published by the Free Software Foundation; either
+  version 2.1 of the License, or (at your option) any later version.
+  The terms of redistributing and/or modifying this software also
+  include exceptions to the LGPL that facilitate static linking.
+
+  This library is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+  Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General Public License
+  along with this library; if not, write to Free Software Foundation, Inc.,
+  51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+*/
+
+#include "teem/nrrd.h"
+#include <testDataPath.h>
+#include <math.h>
+#include <stdlib.h>
+
+/*
+** Tests:
+** nrrdLoad in streaming (i.e., in chunks)
+*/
+
+int
+main(int argc, const char **argv) {
+  const char *me;
+  Nrrd *nin;
+  airArray *mop;
+  char *fullname;
+
+  AIR_UNUSED(argc);
+  me = argv[0];
+  mop = airMopNew();
+
+  nin = nrrdNew();
+  airMopAdd(mop, nin, (airMopper)nrrdNuke, airMopAlways);
+  fullname = testDataPathPrefix("fmob-c4h.nrrd");
+  airMopAdd(mop, fullname, airFree, airMopAlways);
+
+  // Load the header and set header parameters in nin
+  NrrdIoState *nio = nrrdIoStateNew();
+  if (nio == NULL) {
+    fprintf(stderr, "%s: could not allocate NRRD Teem I/O struct \n", me);
+    return 1;
+  }
+
+  nio->skipData = 1;
+
+  if (nrrdLoad(nin, fullname, nio)) {
+    char *err;
+    airMopAdd(mop, err = biffGetDone(NRRD), airFree, airMopAlways);
+    fprintf(stderr, "%s: trouble reading data \"%s\":\n%s",
+            me, fullname, err);
+    airMopError(mop); return 1;
+  }
+
+  free(nio);
+
+  // Create the array where to stream
+  long int numberOfVolumeElement = nrrdElementNumber(nin);
+  double *dataDest = malloc(numberOfVolumeElement*sizeof(*dataDest));
```

Comment:
If we read everything into one big buffer then it somewhat cancels the main advantage that we can read into separate buffers. It would be better to demonstrate how to read individual 3D frames can be read into separate 3D buffers.

Reply:
ok